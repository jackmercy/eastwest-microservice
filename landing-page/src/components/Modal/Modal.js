import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import ReactModal from 'react-modal';

import useStyles from './modal.styles';

const Modal = ({ children, className, open, toggle, ...restProps }) => {
  const classes = useStyles();

  return (
    <ReactModal
      isOpen={open}
      onRequestClose={toggle}
      className={classNames(classes.root, className)}
      overlayClassName={classes.overlay}
      bodyOpenClassName={classes.body}
      ariaHideApp={false}
      {...restProps}
    >
      {children}
    </ReactModal>
  );
};
Modal.propTypes = {
  children: PropTypes.node.isRequired,
  open: PropTypes.bool.isRequired,
  toggle: PropTypes.func,
  className: PropTypes.string,
};
Modal.defaultProps = {
  className: '',
  toggle: undefined,
};

export default memo(Modal);
