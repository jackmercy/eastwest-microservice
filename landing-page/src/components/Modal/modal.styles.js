import { createUseStyles } from 'react-jss';

import COLORS from '../../constants/COLORS';
import { hexToRgbA, rem } from '../../utils/jss';

export default createUseStyles({
  root: {
    top: '10%',
    left: '50%',
    display: 'flex',
    flexDirection: 'column',
    transform: 'translate(-50%, -500px)',
    position: 'absolute',
    overflow: 'auto',
    background: COLORS.bgWhite,
    width: `calc(100% - ${rem(32)})`,
    boxSizing: 'border-box',
    maxWidth: rem(640),
    borderRadius: rem(8),
    maxHeight: '80%',
    boxShadow: '2px 4px 12px rgba(0, 0, 0, 0.15)',
    transition: 'transform 0.5s ease-in-out',
    '&:focus': { outline: 'none' },
    '&.ReactModal__Content--after-open': { transform: 'translate(-50%, 0)' },
  },
  overlay: {
    position: 'fixed',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    zIndex: 900,
    backgroundColor: hexToRgbA(COLORS.bgBlack, 0.1),
    transition: 'background-color .35s linear',
    '&.ReactModal__Overlay--after-open': {
      backgroundColor: hexToRgbA(COLORS.bgBlack, 0.7),
    },
  },
  body: { overflow: 'hidden' },
});
