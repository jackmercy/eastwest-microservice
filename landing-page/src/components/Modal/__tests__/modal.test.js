import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Modal from '../Modal';

describe('Testing Modal component', () => {
  const props = { open: true, toggle: jest.fn() };

  it('should render match snapshot', () => {
    const wrapper = shallow(<Modal {...props}>Modal content</Modal>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
