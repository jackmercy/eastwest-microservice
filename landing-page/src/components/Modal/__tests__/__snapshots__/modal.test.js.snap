// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal component should render match snapshot 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="body-0-2-3"
  className="root-0-2-1"
  closeTimeoutMS={0}
  contentElement={[Function]}
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayClassName="overlay-0-2-2"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  Modal content
</Modal>
`;
